setuptools/monkey.py

```diff
@@ -152,13 +153,13 @@ def patch_for_msvc_specialized_compiler():
     Patch functions in distutils to use standalone Microsoft Visual C++
     compilers.
     """
-    # import late to avoid circular imports on Python < 3.5
-    msvc = import_module('setuptools.msvc')
-
-    if platform.system() != 'Windows':
+    if not get_platform().startswith('win'):
```

Comment:
`platform.system()` just returns Windows on mingw built Python. It's this hack: [Lib/distutils/util.py:42-54](https://github.com/python/cpython/blob/master/Lib/distutils/util.py#L42-L54) which detects it. It seems that running the monkey patch doesn't raise though, so I removed this code.